### ISSUE TITLE
ibm_db.connect line prints "Segmentation fault (core dumped)"  with no Python traceback

### ISSUE THREAD

**Issue by ParfaitG** [ibmdb/python-ibmdb#313]
Python script that used to work previously suddenly now fails with uninformative message which [from this StackOverflow answer](https://stackoverflow.com/a/13654489/1422451) suggests a crash at C level and not in Python:

> Segmentation fault (core dumped)

I tried wrapping `ibm_db.connect` in `try/except` to at least find issue to no avail. I tried uninstalling and reinstalling the `ibm_db` package (even faced [same issue here](https://github.com/ibmdb/python-ibmdb/issues/286) where version 2.0.8a finally worked). Unfortunately, same print line message appears. I tried restarting my machine but still same one line message outputs. Both commented and uncommented `connect` versions below worked seamlessly as seen from my own [SO answer](https://stackoverflow.com/a/47897523/1422451) months ago but now does not. Did something change in my environment to warrant this changed behavior? How can I find the real issue? Is some dump output being generated?

**Resources**

- CPU Environment: Ubuntu 16.04 LTS / Python 3.5 / 64-bit / Intel Core i5 / 8 GB RAM

- DB2 Database: v11.1.2.2 Express-C; resides on another LAN machine from which I am connecting to via Python as a remote client.

**Connect Attempt**

```
import ibm_db, ibm_db_dbi

try:
   #db = ibm_db.connect("Server=127.0.0.1;Port=48000;Hostname=10.0.0.0.1;Database=MYDB;UID=***;PWD=***;", "", "")
   
   # DB CONNECT
   dsn = "DRIVER={IBM DB2 ODBC DRIVER};DATABASE=MYDB;HOSTNAME=10.0.0.0.1;PORT=48000;PROTOCOL=TCPIP;UID=***;PWD=***;"
   db = ibm_db.connect(dsn, "", "")

   conn = ibm_db_dbi.Connection(db)

except Exception as e:
   print(e.message)

finally:
    db = None
    conn = None
```

Output:

` Segmentation fault (core dumped)`

---

Recently, I have been attempting to get DB2 to work with ODBC via unixODBC on Ubuntu to connect to database in R using [CLI/OBDC](http://www-01.ibm.com/support/docview.wss?uid=swg24044630) downloads from IBM (another frustrating situation for another matter). In this process, I have been modifying and adding environment variables . Could this be the issue? Does `ibm_db` rely on any env variable and somehow I broke a link? 

**CLI Driver Environment Variables**:

```
DB2_CLI_DRIVER_INSTALL_PATH=/opt/ibm/odbc_cli/clidriver
LD_LIBRARY_PATH=/opt/oracle/instantclient_12_2:/opt/ibm/odbc_cli/clidriver/lib:
LIBPATH=/opt/ibm//odbc_cli/clidriver/lib
DYLD_LIBRARY_PATH=/opt/ibm//odbc_cli/clidriver/lib
PATH=.../opt/ibm//odbc_cli/clidriver/bin:/opt/ibm//odbc_cli/clidriver/adm...
```

I should add. I do have a Java app that works great with the IBM JDBC driver. So there does not seem to be any issue with my network or connection.

**Comment by SabaKauser** [comment]
Hello @ParfaitG ,

python ibm_db driver does make use of the db2 cli driver. did you set it in IBM_DB_HOME earlier or it was using the one that gets installed with ibm_db installation?
while ibm_db rely on IBM_DB_HOME(and IBM_DB_LIB), db2 cli driver library path gets picked up via LD_LIBRARY_PATH.  
In any case, I would suggest that you collect db2 traces for the problem and share with us.

to collect traces:
cd <clidirver path>/adm
db2trc on -f trc.dmp
<run your application>
db2trc off
db2trc flw trc.dmp trc.flw
db2trc fmt trc.dmp trc.fmt
 and share the trc.flw and trc.fmt files with us to assist you further.

If you can see a non-empty db2dump folder under <clidriver path>/db2dump, kindly share that as well.
I would also like to know the db2level of the cli driver that you are using. (run the exe db2level).

Thanks,
Saba.

**Comment by SabaKauser** [comment]
to collect traces:
cd `clidriver install path`
cd /adm
db2trc on -f trc.dmp
`run your application`
db2trc off
db2trc flw trc.dmp trc.flw
db2trc fmt trc.dmp trc.fmt
and share the trc.flw and trc.fmt files with us to assist you further.

**Comment by ParfaitG** [comment]
@SabaKauser ... you would not believe it. 

Trying `db2trc`, I lreceived
 
> cannot execute binary file: Exec format error

From this I learned I had downloaded the wrong odbc_clidriver for IBM S/390 instead of Linux x86_64. Downloading and extracting that tarball in same location somehow fixed the `ibm_db.connect` issue here! My app now works as usual! Even more I am getting a different error for OBDC connection where I may have to play around with config files. So almost close on that end.

Thanks again for your fast response and attention.
